conan/internal/api/detect_api.py

```diff
@@ -326,21 +326,21 @@ def _gcc_compiler(compiler_exe="gcc"):
             _, out = detect_runner("%s --version" % compiler_exe)
             out = out.lower()
             if "clang" in out:
-                return None, None
```

Comment:
Maybe we can simplify the private ``_xxx_compiler`` and they can still return 2 things, it will be just the root ``detect_compiler`` that will handle it.